src/messages/Message.ts

```diff
@@ -226,6 +251,15 @@ export default class Message {
       offset += message.data.length + 4;
     }
 
+    const messageDetailBytes = byteArrayWithSizeToBytes(data.slice(offset));
```

Comment:
Don't JSON encode this. The purpose of `toBinary` is to keep the data binary.

Use `stringToByteArrayWithSize` for the title and description.
The thumbnail should already be a `Binary`. Prepend the binary with the size.